### PR TITLE
Cover block: fix exception when adding video background

### LINKED ISSUE #53956: Unable to add Video to Cover Block
### Description

When attempting to insert a video into the Cover Block, the video does not display and appears as a blank space both in the editor and black/empty space on the frontend. 

This issue occurs consistently and prevents users from effectively incorporating videos into their content using the Cover Block. The result is consistently a non-functional display.

Other types of media, such as images, seem to load without any problem in the Cover Block. The problem specifically pertains to videos. 

### Step-by-step reproduction instructions

1. Create a new post/page or open an existing one.
2. Add a new block and select the Cover Block.
3. Click on the "Select Media" > "Media Library" in Cover Block settings.
4. Choose the "Video" option and upload/select a video file.
5. Observe that the video space remains blank both in the editor and on the frontend.
6. Attempt to preview or publish the post/page to confirm that the issue persists.

### Screenshots, screen recording, code snippet

Screenshot 2023-08-25 at 12 19 18 PM
Screenshot 2023-08-25 at 12 19 49 PM

https://github.com/WordPress/gutenberg/assets/31700597/404b63b7-203c-40c4-be45-c7a7ab33a79b

 

### Environment info

- WordPress 6.3
- Gutenberg 16.5
- Themes tested: Blank Canvas, Erma

This issue has been tested on multiple instances to ensure that it is not specific to a particular environment. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
Fixes exception that is thrown if video is added as cover background.

## Why?
Adding a video causes fastAverageColor to throw an exception. This was an existing problem but was previously masked by the fact that the exception was thrown in an effect, but now that the check is done in an event handle it disrupts the attribute update process.

Fixes: #53956

## How?
Wraps the call to fastAverageColor in a try/catch and also only passes images to this call, not video.

## Testing Instructions
- Add a Cover block and set a black overlay color, then add text and ensure the text is white
- Reduce the dimRatio below 50 and check that the text turns black
- Increase the dimRatio above 50 again and check that the text turns white again
- Change the overlay color to a light color and check that the text turns black
- Change the dimRatio above and below 50 and check that the text remains black
- Make sure dimRatio is above 50 with the light background and then add a very dark overlay image and check that the text remains black
- Reduce the dimRatio below 50 and check that the text turns white
- Replace the image with a very light one and check that the text turns black
- Remove both the image and the overlay color and check that the text turns white
- Add a new cover block with black overlay color and dimRation < 50
- Add light featured image to the post and toggle it on as the overlay image in the Cover block toolbar and check that the text color is black
- Toggle the feature image back off and check that the text is white again
- Repeat the above two steps with a dark feature image
- Add a Cover block and add a video as the background and make sure it works as expected
- Try replacing the video with another video, and also replacing a video with an image or an image with a video

## Screenshots or screencast 

Before:

https://github.com/WordPress/gutenberg/assets/3629020/1a34b531-744b-4fd9-87ad-83d9dcf76206

After:

https://github.com/WordPress/gutenberg/assets/3629020/e34a5b85-f9c3-46f0-bd1e-cd6323776125

### CODE REVIEW

**Comment by glendaviesnz** [comment]
@ajlende we have been passing video to `retrieveFastAverageColor` all along it seems, but I assume we actually only want to send images? Although the `fast-average-color` lib can handle video I am assuming that due to the time this might take, and the fact that the average of any particular frame could vary radically in relation to any overlayed text, it would be of questionable value anyway.

File: packages/block-library/src/cover/edit/index.js [hunk 1]
```diff
@@ -151,8 +151,9 @@ function CoverEdit( {
 	const setMedia = attributesFromMedia( setAttributes, dimRatio );
 
 	const onSelectMedia = async ( newMedia ) => {
+		const newUrl = newMedia?.type === 'image' ? newMedia.url : undefined;
```
Comment (aaronrobertshaw) [inline packages/block-library/src/cover/edit/index.js:155]: If we don't have a test to cover videos, should we add a small comment here to explain why we're only using any image URL?

**Review by aaronrobertshaw — APPROVED** [review]
Appreciate the quick fix @glendaviesnz 👍 

✅ Could replicate the initial issue

Screenshot 2023-08-26 at 4 19 55 pm


✅ After checking out this PR, I can successfully add a video, with no errors


https://github.com/WordPress/gutenberg/assets/60436221/94729dfd-abbd-4636-8951-290cb95b4304

File: packages/block-library/src/cover/edit/index.js (hunk 1 above)
Comment (glendaviesnz) [reply packages/block-library/src/cover/edit/index.js:155]: Done.

**Comment by glendaviesnz** [comment]
Given that video was just failing by default previously I think this is a reasonable fix. We can follow up if we decide we do want to support video somehow.

**Pull request merged**